document/lib/Pdf/Document/FontDict.hs

```diff
@@ -222,3 +231,85 @@ loadUnicodeCMap pdf fontDict =
             Left e -> throwIO $ Corrupted ("can't parse cmap: " ++ show e) []
             Right cmap -> return $ Just cmap
         _ -> throwIO $ Corrupted "ToUnicode: not a stream" []
+
+
+loadFontDescriptor :: Pdf -> Dict -> IO (Maybe FontDescriptor)
+loadFontDescriptor pdf fontDict = do
+  case HashMap.lookup "FontDescriptor" fontDict of
+    Nothing -> return Nothing
+    Just o -> do
+      ref <- sure $ refValue o
+             `notice` "FontDescriptor should be a reference"
+      fd <- (sure . (`notice` "FontDescriptor: not a dictionary") . dictValue) =<<
+            lookupObject pdf ref
+
+      fontName <- required "FontName" nameValue' fd
+      fontFamily <- optional "FontFamily" (fmap decodeUtf8 . stringValue) fd
+      fontStretch <- optional "FontStretch" nameValue' fd
+      fontWeight <- optional "FontWeight" intValue fd
+      flags <- required "Flags" int64Value fd
+      fontBBox <- optional "FontBBox"
+        (join . fmap (either (const Nothing) Just . rectangleFromArray) . arrayValue) fd
+      italicAngle <- required "ItalicAngle" realValue fd
+      ascent <- optional "Ascent" realValue fd
+      descent <- optional "Descent" realValue fd
+      leading <- optional "Leading" realValue fd
+      capHeight <- optional "CapHeight" realValue fd
+      xHeight <- optional "XHeight" realValue fd
+      stemV <- optional "StemV" realValue fd
+      stemH <- optional "StemH" realValue fd
+      avgWidth <- optional "AvgWidth" realValue fd
+      maxWidth <- optional "MaxWidth" realValue fd
+      missingWidth <- optional "MissingWidth" realValue fd
+      charSet <- optional "CharSet" (fmap decodeUtf8 . stringValue) fd
+
+      return $ Just $ FontDescriptor
+        { fdFontName = fontName
+        , fdFontFamily = fontFamily
+        , fdFontStretch = fontStretch
+        , fdFontWeight = fontWeight
+        , fdFlags = flags
+        , fdFontBBox = fontBBox
+        , fdItalicAngle = italicAngle
+        , fdDescent = descent
+        , fdAscent = ascent
+        , fdLeading = leading
+        , fdCapHeight = capHeight
+        , fdXHeight = xHeight
+        , fdStemV = stemV
+        , fdStemH = stemH
+        , fdAvgWidth = avgWidth
+        , fdMaxWidth = maxWidth
+        , fdMissingWidth = missingWidth
+        , fdCharSet = charSet
+        }
+  where
+    required = requiredInDict "FontDescriptor"
+    optional = optionalInDict "FontDescriptor"
+    nameValue' = fmap (decodeUtf8 . Name.toByteString) . nameValue
```

Comment:
I think names don't have an encoding, probably it's better to keep `FontName` and `FontStretch` as `ByteString`.

Reply:
Yes, I agree. I've changed it.